### PR TITLE
Improve and clean up factory validation exception messages

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Remove old links to Codenvy docs in exception messages
Provide full name (as well as display arrays in it) of the invalid field in exception messages

example of occurring notification in Dashboard:
screenshot from 2018-10-11 16-51-03

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/11562

### CODE REVIEW

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/968/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/968//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11585
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by skabashnyuk — APPROVED** [review]

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/builder/FactoryBuilder.java [hunk 1]
```diff
@@ -287,14 +287,14 @@ void validateCompatibility(
...
+              for (int i = 0; i < list.size(); i++) {
                 validateCompatibility(
-                    entry,
+                    list.get(i),
                     object,
                     secListParamClass,
                     secListParamClass,
                     version,
-                    fullName,
+                    fullName + '[' + i + ']',
```
Comment (mshaposhnik) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/builder/FactoryBuilder.java]: Why do you use `char` quotes?

**Review by mshaposhnik — APPROVED** [review]

**Comment by SkorikSergey** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/968/) doesn't show any regression against this Pull Request.

**Pull request merged**